app/views/govuk_publishing_components/components/_step_by_step_nav.html.erb

```diff
@@ -62,7 +74,18 @@
                 </span>
               </span>
 
-              <span class="js-step-title">
+              <span class="js-step-title"
+                <% if ga4_tracking %>
+                  <% ga4_data = {
+                    event_name: 'select_content',
+                    type: 'step by step',
+                    text: step[:title],
+                    index: step_index + 1,
+                    index_total: steps.length
+                  }.to_json %>
+                  data-ga4="<%= ga4_data %>"
```

Comment:
Note: depending on the outcome of https://github.com/alphagov/govuk_publishing_components/pull/3057 and when it gets merged, this might need to be updated to `data-ga4-event`.